ui/src/app/pages/appAnalysis/apps-list.spec.tsx

```diff
@@ -65,13 +78,53 @@ describe('Apps list', () => {
     expect(startButton(wrapper).prop('disabled')).toBeFalsy();
   });
 
-  it('Clicking Start button should open select application modal that has list of applications', () => {
+  it('should open select application modal that has list of applications on clicking start button', () => {
     const wrapper = component();
     startButton(wrapper).simulate('click');
     expect(
       wrapper.find('[data-test-id="select-application-modal"]')
     ).toBeTruthy();
+    expect(applicationListDropDownWrapper(wrapper).prop('options')).toBe(
+      APP_LIST
+    );
+  });
 
-    expect(applicationListDropDown(wrapper).prop('options')).toBe(APP_LIST);
+  it('should open jupyter modal when Jupyter application is selected and next button is clicked', async () => {
+    const wrapper = component();
+    startButton(wrapper).simulate('click');
+
+    // Application Drop down List should have Jupyter as an option
+    expect(applicationListDropDownWrapper(wrapper).prop('options')).toContain(
+      JUPYTER_APP
+    );
+
+    // Next button should be disabled by default
+    expect(nextButton(wrapper).prop('disabled')).toBe(true);
+
+    await act(async () => {
```

Comment:
When should we use act() ?  I am probably missing some steps in my tests.

Reply:
https://reactjs.org/docs/testing-recipes.html#act
Also i got an error while running test,     _When testing, code that causes React state updates should be wrapped into act(...):_